fastly/block_fastly_service_image_optimizer_default_settings.go

```diff
@@ -0,0 +1,208 @@
+	localState := d.Get(h.Key()).(*schema.Set).List()
+
+	if len(localState) > 0 || d.Get("imported").(bool) || d.Get("force_refresh").(bool) {
+		serviceID := d.Id()
+
+		log.Printf("[DEBUG] Refreshing Image Optimizer default settings for (%s)", serviceID)
+
+		remoteState, err := conn.GetImageOptimizerDefaultSettings(&gofastly.GetImageOptimizerDefaultSettingsInput{
+			ServiceID:      serviceID,
+			ServiceVersion: serviceVersion,
+		})
+		if err != nil {
+			return err
+		}
+
+		result := map[string]any{
+			"allow_video":   remoteState.AllowVideo,
+			"jpeg_type":     remoteState.JpegType,
+			"jpeg_quality":  remoteState.JpegQuality,
+			"resize_filter": remoteState.ResizeFilter,
+			"upscale":       remoteState.Upscale,
+			"webp":          remoteState.Webp,
+			"webp_quality":  remoteState.WebpQuality,
+		}
+
+		results := []map[string]any{result}
+
+		// IMPORTANT: Avoid runtime panic "set item just set doesn't exist".
+		// TF will panic when trying to append an empty map to a TypeSet.
+		// i.e. a typed nil.
+		if len(results[0]) > 0 {
+			if err := d.Set(h.Key(), results); err != nil {
+				log.Printf("[WARN] Error setting Product Enablement for (%s): %s", d.Id(), err)
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// Update updates the resource.
+func (h *ImageOptimizerDefaultSettingsServiceAttributeHandler) Update(_ context.Context, d *schema.ResourceData, _, modified map[string]any, serviceVersion int, conn *gofastly.Client) error {
+	serviceID := d.Id()
+	log.Println("[DEBUG] Update Image Optimizer default settings")
+
+	if len(modified) == 0 {
+		return nil
+	}
+
+	apiInput := gofastly.UpdateImageOptimizerDefaultSettingsInput{
+		ServiceID:      serviceID,
+		ServiceVersion: serviceVersion,
+	}
+
+	for key, value := range modified {
+		switch key {
+		case "resize_filter":
+			var resizeFilter gofastly.ResizeFilter
+			switch value.(string) {
+			case "lanczos3":
+				resizeFilter = gofastly.Lanczos3
+			case "lanczos2":
+				resizeFilter = gofastly.Lanczos2
+			case "bicubic":
+				resizeFilter = gofastly.Bicubic
+			case "bilinear":
+				resizeFilter = gofastly.Bilinear
+			case "nearest":
+				resizeFilter = gofastly.Nearest
+			default:
+				return fmt.Errorf("got unexpected resize_filter: %v", value)
+			}
+			apiInput.ResizeFilter = &resizeFilter
+		case "webp":
+			webp := value.(bool)
+			apiInput.Webp = &webp
+		case "webp_quality":
+			webpQuality := value.(int)
+			apiInput.WebpQuality = &webpQuality
+		case "jpeg_type":
+			jpegType := value.(string)
+			apiInput.JpegType = &jpegType
+		case "jpeg_quality":
+			jpegQuality := value.(int)
+			apiInput.JpegQuality = &jpegQuality
+		case "upscale":
+			upscale := value.(bool)
+			apiInput.Upscale = &upscale
+		case "allow_video":
+			allowVideo := value.(bool)
+			apiInput.AllowVideo = &allowVideo
```

Comment:
Everywhere you've created a variable for the sake of taking its address you can instead use `gofastly.ToPointer(value.(whatever))`.

Reply:
Oh awesome! That's much better haha.